internal/subscription/core.go

```diff
@@ -497,3 +501,60 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
 	logger.Ctx(ctx).Infow("migration: request completed.", "subscriptionsUpdated", updatedSubCount)
 	return nil
 }
+
+func (c *Core) deleteSubscriptionTopics(ctx context.Context, m *Model) error {
+	if m == nil || m.GetTopic() == "" {
+		return errors.New("Nil model")
+	}
+
+	projectID := m.ExtractedTopicProjectID
+
+	//  delete internal topic
+	internalTopic := m.GetSubscriptionTopic()
+	err := c.deleteSubscriptionTopic(ctx, internalTopic, projectID, topic.GetTopicNameOnly(internalTopic))
+	if err != nil {
+		return err
+	}
+
+	// delete retry topic
+	retryTopic := m.GetRetryTopic()
+	err = c.deleteSubscriptionTopic(ctx, retryTopic, projectID, topic.GetTopicNameOnly(retryTopic))
+	if err != nil {
+		return err
+	}
+
+	// delete delay topics
+	for _, delayTopic := range m.GetDelayTopics() {
+		delayModel := &topic.Model{
+			Name:               delayTopic,
+			ExtractedProjectID: projectID,
+			ExtractedTopicName: topic.GetTopicNameOnly(delayTopic),
+		}
+		err = c.topicCore.DeleteSubscriptionTopic(ctx, delayModel)
+		if err != nil {
+			return err
+		}
+	}
+
+	// since dlq topic is not created for dlq subscription no need to delete dlq topic
+	if m.IsDeadLetterSubscription() {
+		return nil
+	}
+
+	// delete dlq topics
+	dlqTopic := m.GetDeadLetterTopic()
+	err = c.deleteSubscriptionTopic(ctx, dlqTopic, projectID, topic.GetTopicNameOnly(dlqTopic))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Core) deleteSubscriptionTopic(ctx context.Context, name, projectID, extractedTopicName string) error {
+	return c.topicCore.DeleteSubscriptionTopic(ctx, &topic.Model{
```

Comment:
Can this be made a bulk operation rather than multiple invocations?

Reply:
deleted  the mathod, made changes in deleteSubscription topics